pkg/vulnsrc/osv/osv.go

```diff
@@ -0,0 +1,223 @@
+	for _, osv := range osvs {
+		vulnId := getVulnId(&osv)
+
+		for _, affected := range osv.Affected {
+			var vulnerableVersions []string
+			var patchedVersions []string
+
+			firstVersion := vs.ecosystem.firstVersion
+			if len(affected.Versions) != 0 && affected.Versions[0] != "" {
+				firstVersion = affected.Versions[0]
+			}
+
+			for _, rng := range affected.Ranges {
+				if rng.Type == vs.ecosystem.eventType {
+					var vulnerableVersion string
+
+					for _, event := range rng.Events {
+						if event.Fixed != "" && event.Introduced != "" { //fixed and introduced in 1 struct
+							patchedVersions = append(patchedVersions, event.Fixed)
+							vulnerableVersions = append(vulnerableVersions, fmt.Sprintf(">=%s <%s", event.Introduced, event.Fixed))
+						} else if event.Introduced != "" {
+							if vulnerableVersion != "" {
+								vulnerableVersions = append(vulnerableVersions, vulnerableVersion)
+							}
+							vulnerableVersion = fmt.Sprintf(">=%s", event.Introduced)
+						} else if event.Fixed != "" {
+							if vulnerableVersion != "" {
+								vulnerableVersion = vulnerableVersion + fmt.Sprintf(" <%s", event.Fixed)
+							} else {
+								vulnerableVersion = fmt.Sprintf(">=%s <%s", firstVersion, event.Fixed)
+							}
+							patchedVersions = append(patchedVersions, event.Fixed)
+						}
+					}
+					if vulnerableVersion != "" {
+						vulnerableVersions = append(vulnerableVersions, vulnerableVersion)
+					}
+				}
+			}
+
+			advisory := types.Advisory{
+				VulnerableVersions: vulnerableVersions,
+				PatchedVersions:    patchedVersions,
+			}
+
+			if err := vs.dbc.PutAdvisoryDetail(tx, vulnId, fmt.Sprintf(platformFormat, vs.ecosystem.name), affected.Package.Name, advisory); err != nil {
+				return xerrors.Errorf("failed to save osv advisory: %w", err)
+			}
+
+			var references []string
+			for _, ref := range osv.References {
+				references = append(references, ref.Url)
+			}
+
+			vuln := types.VulnerabilityDetail{
+				ID:               vulnId,
+				Description:      osv.Details,
+				PublishedDate:    MustParse(time.RFC3339, osv.Published),
+				LastModifiedDate: MustParse(time.RFC3339Nano, osv.Modified),
+				Title:            osv.Id,
+				References:       references,
+			}
+
+			if err := vs.dbc.PutVulnerabilityDetail(tx, vulnId, fmt.Sprintf(datasourceFormat, strings.ToLower(vs.ecosystem.name)), vuln); err != nil {
+				return xerrors.Errorf("failed to save osv vulnerability: %w", err)
+			}
+
+			// for light DB
+			if err := vs.dbc.PutSeverity(tx, vulnId, types.SeverityUnknown); err != nil {
+				return xerrors.Errorf("failed to save osv vulnerability severity for light: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+func getEcoSystem(ecosystemName string) ecosystem {
+	for _, system := range defaultEcosystem {
+		if system.name == ecosystemName {
+			return system
+		}
+	}
+	return ecosystem{}
+}
+
+func (vs VulnSrc) Get(pkgName string) ([]types.Advisory, error) {
```

Comment:
this is very similar to
https://github.com/aquasecurity/trivy-db/blob/main/pkg/db/advisory.go#L23

Reply:
used `GetAdvisories`

Reply:
removed `Get()`